Dockerfile

```diff
@@ -0,0 +1,20 @@
+# Alpine for smaller size
+FROM python:3.9-alpine
+
+# Create a system account hubibot.hubibot
+RUN addgroup -S hubibot && adduser -S hubibot -G hubibot
+# Non-alpine equivalent of above:
+#RUN groupadd -r hubibot && useradd -r -m -g hubibot hubibot
+
+USER hubibot
+
+WORKDIR /app
+
+COPY requirements.txt     /app
+COPY main.py              /app
+COPY template.config.yaml /app
+
+RUN pip install --no-cache-dir --upgrade pip && \
```

Comment:
Consider re-arrange lines a little bit - first copy requirements, than installing requirements, later - copy the rest.

Docker caches files, so if a requirement file has not changed, the cache requirements will be installed.

In the way the code is written right now requirements are installed every time main.py is changed, that I assume is pretty often.

Reply:
done